[textDocument/codeAction] [LSP] System.ArgumentException: Range={ Start={ Line=7, Character=0 }, End={ Line=7, Character=2147483647 } }.

Type: <b>Bug</b>

## Issue Description ##
```
2026-04-22 21:26:13.591 [error] [textDocument/codeAction] [LSP] System.ArgumentException: Range={ Start={ Line=7, Character=0 }, End={ Line=7, Character=2147483647 } }. text.Length=7074. text.Lines.Count=146
 ---> System.ArgumentOutOfRangeException: 'end' must not be less than 'start'. start='537' end='-2147483112'. (Parameter 'end')
   at Microsoft.CodeAnalysis.LanguageServer.ProtocolConversions.RangeToTextSpan(Range range, SourceText text) in /_/src/LanguageServer/Protocol/Extensions/ProtocolConversions.cs:line 336
   --- End of inner exception stack trace ---
   at Microsoft.CodeAnalysis.LanguageServer.ProtocolConversions.RangeToTextSpan(Range range, SourceText text) in /_/src/LanguageServer/Protocol/Extensions/ProtocolConversions.cs:line 341
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions.CodeActionHelpers.GetActionSetsAsync(TextDocument document, ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService, Range selection, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:line 389
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions.CodeActionHelpers.GetVSCodeActionsAsync(CodeActionParams request, TextDocument document, ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService, Boolean hasVsLspCapability, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:line 42
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActionsHandler.HandleRequestAsync(CodeActionParams request, RequestContext context, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs:line 57
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`1.StartRequestAsync[TRequest,TResponse](TRequest request, TRequestContext context, IMethodHandler handler, String language, CancellationToken cancellationToken)
```
## Steps to Reproduce ##

## Expected Behavior ##

## Actual Behavior ##



Extension version: 2.130.5
VS Code version: Code 1.116.0 (Universal) (560a9dba96f961efea7b1612916f89e5d5d4d679, 2026-04-15T00:28:13Z)
OS version: Darwin arm64 25.3.0
Modes:

### Logs ###

<!--
If you can, it would be the most helpful to zip up and attach the entire extensions log folder.  The folder can be opened by running the `workbench.action.openExtensionLogsFolder` command.

Additionally, if you can reproduce the issue reliably, use trace logging, see https://github.com/dotnet/vscode-csharp/blob/main/SUPPORT.md#c-lsp-trace-logs and then report the issue.
-->
### C# log ###
<details>

```
2026-04-22 18:46:44.399 [info] Locating .NET runtime version 10.0.5
2026-04-22 18:46:44.466 [info] Dotnet path: /usr/local/share/dotnet/dotnet
2026-04-22 18:46:44.466 [info] Activating C# + C# Dev Kit...
2026-04-22 18:46:45.112 [info] info: Program[0]
2026-04-22 18:46:45.112 [info]       Server started with process ID 90184
2026-04-22 18:46:45.135 [info] [Program] Language server initialized
2026-04-22 18:46:45.140 [info] [Program] Monitoring client process 90097 for exit
2026-04-22 18:46:47.436 [info] [textDocument/_vs_getProjectContexts] [Microsoft.CodeAnalysis.MSBuild.BuildHostProcessManager] .NET BuildHost started from /Users/harshs/.vscode/extensions/ms-dotnettools.csharp-2.130.5-darwin-arm64/.roslyn/Microsoft.CodeAnalysis.LanguageServer reloading to start from /usr/local/share/dotnet/dotnet to match necessary SDK location.
2026-04-22 18:46:48.144 [warning] [textDocument/_vs_getProjectContexts] [LanguageServerProjectLoader] Project /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/dd4c6bac-cbcf-4f1f-ac6e-16b0029d9c13/Canonical.csproj has unresolved dependencies
2026-04-22 18:46:48.211 [info] [textDocument/_vs_getProjectContexts] [LanguageServerProjectLoader] Successfully completed load of /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/dd4c6bac-cbcf-4f1f-ac6e-16b0029d9c13/Canonical.cs
2026-04-22 18:46:48.213 [info] [textDocument/_vs_getProjectContexts] [LanguageServerProjectLoader] Restoring Canonical.cs: Running dotnet restore on /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/dd4c6bac-cbcf-4f1f-ac6e-16b0029d9c13/Canonical.cs
2026-04-22 18:46:48.213 [info] [textDocument/_vs_getProjectContexts] [Microsoft.CodeAnalysis.LanguageServer.DotnetCliHelper] Using dotnet executable configured on the PATH
2026-04-22 18:46:48.622 [info] [serviceBroker/connect] [WorkspaceProjectFactoryService] Project /Users/harshs/Documents/Micrologic/Gitlab/LogicWash.UI/LogicWash.UI.csproj loaded by C# Dev Kit
2026-04-22 18:46:48.627 [info] [serviceBroker/connect] [WorkspaceProjectFactoryService] Project /Users/harshs/Documents/Micrologic/Gitlab/LogicWash.UI/LogicWash.UI.csproj loaded by C# Dev Kit
2026-04-22 18:46:48.697 [info] [textDocument/_vs_getProjectContexts] [LanguageServerProjectLoader] Restoring Canonical.cs:   Determining projects to restore...
2026-04-22 18:46:50.822 [info] [textDocument/_vs_getProjectContexts] [LanguageServerProjectLoader] Restoring Canonical.cs: /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/dd4c6bac-cbcf-4f1f-ac6e-16b0029d9c13/Canonical.csproj : warning NU1900: Error occurred while getting package vulnerability data: Access to the path '/Users/harshs/.local/share/NuGet/http-cache' is denied.
2026-04-22 18:46:50.856 [info] [textDocument/_vs_getProjectContexts] [LanguageServerProjectLoader] Restoring Canonical.cs:   Restored /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/dd4c6bac-cbcf-4f1f-ac6e-16b0029d9c13/Canonical.csproj (in 2.05 sec).
2026-04-22 18:46:50.871 [info] [textDocument/_vs_getProjectContexts] [LanguageServerProjectLoader] Restore complete
2026-04-22 18:46:50.871 [info] [textDocument/_vs_getProjectContexts] [LanguageServerProjectLoader] Completed (re)load of all projects in 00:00:03.6804290
2026-04-22 21:26:13.591 [error] [textDocument/codeAction] [LSP] System.ArgumentException: Range={ Start={ Line=7, Character=0 }, End={ Line=7, Character=2147483647 } }. text.Length=7074. text.Lines.Count=146
 ---> System.ArgumentOutOfRangeException: 'end' must not be less than 'start'. start='537' end='-2147483112'. (Parameter 'end')
   at Microsoft.CodeAnalysis.LanguageServer.ProtocolConversions.RangeToTextSpan(Range range, SourceText text) in /_/src/LanguageServer/Protocol/Extensions/ProtocolConversions.cs:line 336
   --- End of inner exception stack trace ---
   at Microsoft.CodeAnalysis.LanguageServer.ProtocolConversions.RangeToTextSpan(Range range, SourceText text) in /_/src/LanguageServer/Protocol/Extensions/ProtocolConversions.cs:line 341
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions.CodeActionHelpers.GetActionSetsAsync(TextDocument document, ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService, Range selection, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:line 389
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions.CodeActionHelpers.GetVSCodeActionsAsync(CodeActionParams request, TextDocument document, ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService, Boolean hasVsLspCapability, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:line 42
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActionsHandler.HandleRequestAsync(CodeActionParams request, RequestContext context, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs:line 57
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`1.StartRequestAsync[TRequest,TResponse](TRequest request, TRequestContext context, IMethodHandler handler, String language, CancellationToken cancellationToken)
2026-04-22 21:26:13.592 [error] Request textDocument/codeAction failed.
  Message: Range={ Start={ Line=7, Character=0 }, End={ Line=7, Character=2147483647 } }. text.Length=7074. text.Lines.Count=146
  Code: -32000 
[object Object]
2026-04-22 21:26:18.319 [error] [textDocument/codeAction] [LSP] System.ArgumentException: Range={ Start={ Line=7, Character=0 }, End={ Line=7, Character=2147483647 } }. text.Length=7074. text.Lines.Count=146
 ---> System.ArgumentOutOfRangeException: 'end' must not be less than 'start'. start='537' end='-2147483112'. (Parameter 'end')
   at Microsoft.CodeAnalysis.LanguageServer.ProtocolConversions.RangeToTextSpan(Range range, SourceText text) in /_/src/LanguageServer/Protocol/Extensions/ProtocolConversions.cs:line 336
   --- End of inner exception stack trace ---
   at Microsoft.CodeAnalysis.LanguageServer.ProtocolConversions.RangeToTextSpan(Range range, SourceText text) in /_/src/LanguageServer/Protocol/Extensions/ProtocolConversions.cs:line 341
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions.CodeActionHelpers.GetActionSetsAsync(TextDocument document, ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService, Range selection, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:line 389
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions.CodeActionHelpers.GetVSCodeActionsAsync(CodeActionParams request, TextDocument document, ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService, Boolean hasVsLspCapability, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs:line 42
   at Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActionsHandler.HandleRequestAsync(CodeActionParams request, RequestContext context, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs:line 57
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`1.StartRequestAsync[TRequest,TResponse](TRequest request, TRequestContext context, IMethodHandler handler, String language, CancellationToken cancellationToken)
2026-04-22 21:26:18.320 [error] Request textDocument/codeAction failed.
  Message: Range={ Start={ Line=7, Character=0 }, End={ Line=7, Character=2147483647 } }. text.Length=7074. text.Lines.Count=146
  Code: -32000 
[object Object]
2026-04-22 21:26:36.520 [info] Locating .NET runtime version 10.0.5
2026-04-22 21:26:36.566 [info] Dotnet path: /usr/local/share/dotnet/dotnet
2026-04-22 21:26:36.566 [info] Activating C# + C# Dev Kit...
2026-04-22 21:26:36.639 [info] info: Program[0]
2026-04-22 21:26:36.639 [info]       Server started with process ID 17666
2026-04-22 21:26:36.915 [info] [Program] Language server initialized
2026-04-22 21:26:36.917 [info] [Program] Monitoring client process 17594 for exit
2026-04-22 21:29:41.182 [info] Locating .NET runtime version 10.0.5
2026-04-22 21:29:41.261 [info] Dotnet path: /usr/local/share/dotnet/dotnet
2026-04-22 21:29:41.261 [info] Activating C# + C# Dev Kit...
2026-04-22 21:29:41.403 [info] info: Program[0]
2026-04-22 21:29:41.403 [info]       Server started with process ID 19445
2026-04-22 21:29:42.964 [info] [Program] Language server initialized
2026-04-22 21:29:42.979 [info] [Program] Monitoring client process 19323 for exit
2026-04-22 21:29:44.416 [info] [serviceBroker/connect] [WorkspaceProjectFactoryService] Project /Users/harshs/Documents/Micrologic/Gitlab/LogicWash.UI/LogicWash.UI.csproj loaded by C# Dev Kit
2026-04-22 21:29:44.495 [info] [serviceBroker/connect] [WorkspaceProjectFactoryService] Project /Users/harshs/Documents/Micrologic/Gitlab/LogicWash.UI/LogicWash.UI.csproj loaded by C# Dev Kit
2026-04-22 22:59:03.493 [error] Unhandled exception. System.ObjectDisposedException: Cannot access a disposed object.
2026-04-22 22:59:03.494 [error] Object name: 'System.Net.Security.SslStream'.
2026-04-22 22:59:03.494 [error]    at System.Net.Security.SslStream.<ThrowIfExceptional>g__ThrowExceptional|131_0(ExceptionDispatchInfo e)
2026-04-22 22:59:03.494 [error]    at System.Net.Security.SslStream.WriteAsyncInternal[TIOAdapter](ReadOnlyMemory`1 buffer, CancellationToken cancellationToken)
2026-04-22 22:59:03.494 [error]    at System.Net.Security.SslStream.Write(Byte[] buffer, Int32 offset, Int32 count)
2026-04-22 22:59:03.494 [error]    at Microsoft.VisualStudio.DesignTools.RuntimeHost.Networking.TcpNetworkStream.Write(Byte[] buffer)
2026-04-22 22:59:03.494 [error]    at Microsoft.VisualStudio.DesignTools.RuntimeHost.Networking.TcpDataBridge.WriteMessage(Byte[] buffer)
2026-04-22 22:59:03.494 [error]    at Microsoft.VisualStudio.DesignTools.RuntimeHost.Networking.ProtocolHandler.<>c__DisplayClass48_0.<PushMessageOnToSendQueue>b__0()
2026-04-22 22:59:03.494 [error]    at Microsoft.VisualStudio.DesignTools.RuntimeHost.Networking.ProtocolHandler.ActionThread(Object list)
2026-04-22 22:59:03.494 [error]    at System.Threading.Thread.StartHelper.Callback(Object state)
2026-04-22 22:59:03.494 [error]    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
2026-04-22 22:59:03.494 [error] --- End of stack trace from previous location ---
2026-04-22 22:59:03.494 [error]    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
2026-04-22 22:59:03.494 [error]    at System.Threading.Thread.StartCallback()
2026-04-22 22:59:03.574 [info] Connection to server got closed. Server will restart.
2026-04-22 22:59:03.575 [error] Server process exited with signal SIGABRT.
2026-04-22 22:59:03.659 [info] info: Program[0]
2026-04-22 22:59:03.659 [info]       Server started with process ID 34210
2026-04-22 22:59:03.914 [info] [Program] Language server initialized
2026-04-22 22:59:03.916 [info] [Program] Monitoring client process 19323 for exit
2026-04-22 22:59:04.415 [info] [textDocument/documentSymbol] [Microsoft.CodeAnalysis.MSBuild.BuildHostProcessManager] .NET BuildHost started from /Users/harshs/.vscode/extensions/ms-dotnettools.csharp-2.130.5-darwin-arm64/.roslyn/Microsoft.CodeAnalysis.LanguageServer reloading to start from /usr/local/share/dotnet/dotnet to match necessary SDK location.
2026-04-22 22:59:04.881 [warning] [textDocument/documentSymbol] [LanguageServerProjectLoader] Project /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/990ce5b5-4fa1-45b7-bc72-c2d9fa125f74/Canonical.csproj has unresolved dependencies
2026-04-22 22:59:04.942 [info] [textDocument/documentSymbol] [LanguageServerProjectLoader] Successfully completed load of /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/990ce5b5-4fa1-45b7-bc72-c2d9fa125f74/Canonical.cs
2026-04-22 22:59:04.944 [info] [textDocument/documentSymbol] [LanguageServerProjectLoader] Restoring Canonical.cs: Running dotnet restore on /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/990ce5b5-4fa1-45b7-bc72-c2d9fa125f74/Canonical.cs
2026-04-22 22:59:04.944 [info] [textDocument/documentSymbol] [Microsoft.CodeAnalysis.LanguageServer.DotnetCliHelper] Using dotnet executable configured on the PATH
2026-04-22 22:59:05.353 [info] [textDocument/documentSymbol] [LanguageServerProjectLoader] Restoring Canonical.cs:   Determining projects to restore...
2026-04-22 22:59:07.641 [info] [textDocument/documentSymbol] [LanguageServerProjectLoader] Restoring Canonical.cs: /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/990ce5b5-4fa1-45b7-bc72-c2d9fa125f74/Canonical.csproj : warning NU1900: Error occurred while getting package vulnerability data: Access to the path '/Users/harshs/.local/share/NuGet/http-cache' is denied.
2026-04-22 22:59:07.676 [info] [textDocument/documentSymbol] [LanguageServerProjectLoader] Restoring Canonical.cs:   Restored /var/folders/rx/xb_kzkx17jv6cc_088gtqkp80000gq/T/roslyn-canonical-misc/990ce5b5-4fa1-45b7-bc72-c2d9fa125f74/Canonical.csproj (in 2.22 sec).
2026-04-22 22:59:07.693 [info] [textDocument/documentSymbol] [LanguageServerProjectLoader] Restore complete
2026-04-22 22:59:07.693 [info] [textDocument/documentSymbol] [LanguageServerProjectLoader] Completed (re)load of all projects in 00:00:03.4027920

```
</details>

### C# LSP Trace Logs log ###
<details>

```

```
</details>



## Environment Information ##
**VSCode version**: 1.116.0
**C# Extension**: 2.130.5
**Using OmniSharp**: false


<details><summary>Dotnet Information</summary>
.NET SDK:
 Version:           10.0.102
 Commit:            4452502459
 Workload version:  10.0.105
 MSBuild version:   18.0.7+445250245

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  26.3
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/10.0.102/

.NET workloads installed:
 [maui]
   Installation Source: SDK 10.0.100
   Manifest Version:    10.0.20/10.0.100
   Manifest Path:       /usr/local/share/dotnet/sdk-manifests/10.0.100/microsoft.net.sdk.maui/10.0.20/WorkloadManifest.json
   Install Type:        FileBased

Configured to use workload sets when installing new manifests.

Host:
  Version:      10.0.5
  Architecture: arm64
  Commit:       a612c2a105

.NET SDKs installed:
  10.0.100 [/usr/local/share/dotnet/sdk]
  10.0.102 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 10.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 10.0.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 10.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 10.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 10.0.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 10.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_CLI_UI_LANGUAGE                   [en-US]
  DOTNET_MULTILEVEL_LOOKUP                 [0]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|Folder Name|
|---|---|---|---|
|amazon-q-vscode|amazonwebservices|2.0.0|amazonwebservices.amazon-q-vscode-2.0.0|
|csdevkit|ms-dotnettools|3.10.14|ms-dotnettools.csdevkit-3.10.14-darwin-arm64|
|csharp|ms-dotnettools|2.130.5|ms-dotnettools.csharp-2.130.5-darwin-arm64|
|dotnet-maui|ms-dotnettools|1.14.10|ms-dotnettools.dotnet-maui-1.14.10-darwin-arm64|
|dotnet-meteor|nromanov|6.2.9|nromanov.dotnet-meteor-6.2.9-darwin-arm64|
|githistory|donjayamanne|0.6.20|donjayamanne.githistory-0.6.20|
|kiro-theme|MohdZaid|1.0.0|mohdzaid.kiro-theme-1.0.0|
|vscode-dotnet-runtime|ms-dotnettools|3.0.0|ms-dotnettools.vscode-dotnet-runtime-3.0.0|
|xamlstyler|dabbinavo|1.6.0|dabbinavo.xamlstyler-1.6.0|;

</details>
<details><summary>C# Settings</summary>

|Setting|Value|
|---|---|
|preferCSharpExtension|false|
|compilerDiagnosticScope|openFiles|
|analyzerDiagnosticScope|openFiles|
|enableXamlTools|true|
|useServerGC|true|;

</details>
<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M5 (10 x 2400)|
|GPU Status|2d_canvas: enabled<br>GPU0: VENDOR= 0x106b [Google Inc. (Apple)], DEVICE=0x0000 [ANGLE (Apple, ANGLE Metal Renderer: Apple M5, Version 26.3.1 (a) (Build 25D771280a))], DRIVER_VENDOR=Apple, DRIVER_VERSION=26.3.1 *ACTIVE*<br>Machine model name: Mac<br>Machine model version: 17.2<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: enabled_on<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|5, 4, 3|
|Memory (System)|32.00GB (2.38GB free)|
|Process Argv|--crash-reporter-id 4a73fb46-2ae9-4ec0-a9da-8f8da859a2ae|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vswsl492:30256859
binariesv615:30325510
nativeloc1:31344060
dwcopilot:31170013
dwoutputs:31242946
copilot_t_ci:31333650
e5gg6876:31282496
pythonrdcb7:31342333
6518g693:31463988
aj953862:31281341
6abeh943:31336334
envsactivate1:31464700
cloudbuttont:31379625
3efgi100_wstrepl:31403338
use-responses-api:31390855
je187915:31454425
ec5jj548:31422691
cp_cls_t_966_ss:31454198
inlinechat_v2_hd992725:31499106
8hhj4413:31478653
38bie571_auto:31426784
cp_cls_c_1081:31454833
conptydll_true:31498968
ia-use-proxy-models-svc:31452481
e9c30283:31461165
test_treatment2:31471001
control_6dc23131:31497324
idci7584:31464702
e3e4d672:31494082
ei9d7968:31496641
534a6447:31496642
chat:31457767
8hig5102:31480529
67jbj424:31495046
7e187181:31499614
cpptoolsoff-v2:31475362
i2gc6536:31499202
1g3cj959:31499302
ghj88844:31499326
ec1eg703:31491645
client_tst_t:31495907
ddid_t:31478206
getcmakediagnosticsoff:31489825
g_20_d1382936:31495671
nes-perm-reb-0:31490411
cp_cls_c_1082:31491634
719di409_sum_t:31499345
logging_enabled_new:31498466
j0d79568:31499440
56dj4588:31497896

```

</details>

<!-- generated by issue reporter -->

--- a/src/LanguageServer/Protocol/Extensions/ProtocolConversions.cs
+++ b/src/LanguageServer/Protocol/Extensions/ProtocolConversions.cs
@@ -326,6 +326,10 @@ internal static partial class ProtocolConversions
     {
         var linePositionSpan = RangeToLinePositionSpan(range);
 
+        linePositionSpan = new LinePositionSpan(
+            ClampPositionToLineEnd(linePositionSpan.Start, text),
+            ClampPositionToLineEnd(linePositionSpan.End, text));
+
         // Handle the specific case where the end position is exactly one line beyond the document bounds
         // and the end character is 0 (start of the non-existent next line).
         // This can happen when deleting the last line, where LSP clients are allowed (by the spec) to
@@ -363,6 +367,18 @@ internal static partial class ProtocolConversions
 
         static string PositionToString(LSP.Position position)
             => $"{{ Line={position.Line}, Character={position.Character} }}";
+
+        static LinePosition ClampPositionToLineEnd(LinePosition position, SourceText text)
+        {
+            if (position.Line < 0 || position.Line >= text.Lines.Count)
+                return position;
+
+            var line = text.Lines[position.Line];
+            var lineLength = line.End - line.Start;
+            return position.Character > lineLength
+                ? new LinePosition(position.Line, lineLength)
+                : position;
+        }
     }
 
     public static LSP.TextEdit TextChangeToTextEdit(TextChange textChange, SourceText oldText)

--- a/src/LanguageServer/ProtocolUnitTests/ProtocolConversionsTests.cs
+++ b/src/LanguageServer/ProtocolUnitTests/ProtocolConversionsTests.cs
@@ -236,6 +236,32 @@ public sealed class ProtocolConversionsTests : AbstractLanguageServerProtocolTes
     }
 
     [Fact]
+    public void RangeToTextSpanClampsCharacterPastLineEnd()
+    {
+        var markup = GetTestMarkup();
+        var sourceText = SourceText.From(markup);
+
+        var range = new Range() { Start = new Position(2, 8), End = new Position(2, int.MaxValue) };
+        var textSpan = ProtocolConversions.RangeToTextSpan(range, sourceText);
+
+        Assert.Equal(21, textSpan.Start);
+        Assert.Equal(27, textSpan.End);
+    }
+
+    [Fact]
+    public void RangeToTextSpanClampsStartCharacterPastLineEnd()
+    {
+        var markup = GetTestMarkup();
+        var sourceText = SourceText.From(markup);
+
+        var range = new Range() { Start = new Position(2, int.MaxValue), End = new Position(2, int.MaxValue) };
+        var textSpan = ProtocolConversions.RangeToTextSpan(range, sourceText);
+
+        Assert.Equal(27, textSpan.Start);
+        Assert.Equal(27, textSpan.End);
+    }
+
+    [Fact]
     public void RangeToTextSpanLineEndOfDocument()
     {
         var markup = GetTestMarkup();
@@ -321,8 +347,8 @@ public sealed class ProtocolConversionsTests : AbstractLanguageServerProtocolTes
         var markup = GetTestMarkup();
         var sourceText = SourceText.From(markup);
 
-        // This start position will be beyond the end position
-        var range = new Range() { Start = new Position(2, 20), End = new Position(3, 0) };
+        // This start position will still be beyond the end position after clamping.
+        var range = new Range() { Start = new Position(2, int.MaxValue), End = new Position(2, 0) };
         Assert.Throws<ArgumentException>(() => ProtocolConversions.RangeToTextSpan(range, sourceText));
     }
 

--- a/src/Razor/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/SourceTextExtensions.cs
+++ b/src/Razor/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/SourceTextExtensions.cs
@@ -235,9 +235,9 @@ internal static class SourceTextExtensions
 
         var textLine = text.Lines[line];
 
-        if (character > textLine.SpanIncludingLineBreak.Length)
+        if (character > textLine.Span.Length)
         {
-            return false;
+            character = textLine.Span.Length;
         }
 
         absoluteIndex = textLine.Start + character;

--- a/src/Razor/src/Razor/test/Microsoft.CodeAnalysis.Razor.Workspaces.UnitTests/Extensions/SourceTextExtensionsTest.cs
+++ b/src/Razor/src/Razor/test/Microsoft.CodeAnalysis.Razor.Workspaces.UnitTests/Extensions/SourceTextExtensionsTest.cs
@@ -1,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Razor.Workspaces.Test.Extensions;
+
+public class SourceTextExtensionsTest
+{
+    [Fact]
+    public void TryGetAbsoluteIndex_ClampsCharacterPastLineEnd()
+    {
+        var sourceText = SourceText.From("hello\r\nworld");
+
+        var result = sourceText.TryGetAbsoluteIndex(line: 0, character: int.MaxValue, out var absoluteIndex);
+
+        Assert.True(result);
+        Assert.Equal(5, absoluteIndex);
+    }
+
+    [Fact]
+    public void TryGetAbsoluteIndex_ClampsCharacterInsideLineBreakToLineEnd()
+    {
+        var sourceText = SourceText.From("hello\r\nworld");
+
+        var result = sourceText.TryGetAbsoluteIndex(line: 0, character: 6, out var absoluteIndex);
+
+        Assert.True(result);
+        Assert.Equal(5, absoluteIndex);
+    }
+
+    [Fact]
+    public void GetTextSpan_ClampsCharactersPastLineEnd()
+    {
+        var sourceText = SourceText.From("hello\r\nworld");
+
+        var textSpan = sourceText.GetTextSpan(startLine: 0, startCharacter: 0, endLine: 0, endCharacter: int.MaxValue);
+
+        Assert.Equal(new TextSpan(0, 5), textSpan);
+    }
+}

